SA/design_concept/content/results/outlook_ide.typ

```diff
@@ -0,0 +1,32 @@
+= IDE for visual functional programming
+In a future project, VisualFP could be further enhanced to work like an IDE.
+Besides the functionality that already exists in the PoC application, this would require:
+- An editor for custom types
+- A way to organize defined functions & types
+- Possibility to save & open defined functions & types
+- Mean to execute functions
+- Haskell compatibility
+
+@visualfp_ide_mockup shows a mockup of how the VisualFP IDE could look like.
+The two main elements are the sidebar, which gives the user access to pre- and self-defined functions, and the large editor-section, where functions & data types can be defined visually.
+
+Functions in the sidebar are grouped into modules (analogous to Haskell modules), which appear like folders in the IDE.
+The user has the possibility to create their own modules and to import modules from the prelude or external packages.
+
+#figure(
+    image("../../static/visualfp_ide_mockup.png"),
+    caption: "Mockup of VisualFP IDE"
+)<visualfp_ide_mockup>
+
+Importing external packages requires VisualFP to be compatible with Haskell.
+For the type inference to work, a future project would need to find a way to get the type signature of Haskell functions and convert it to VisualFP's type model.
+
+Regarding code execution it is possible to convert VisualFP's value definition to Haskell.
```

Comment:
I think that is clear. Maybe we could write something about compatibility with Haskell debugging tools?

We are free to dream here, after all 😅